pico_project.py

```diff
@@ -823,64 +711,29 @@ def ParseCommandLine():
 
 def GenerateMain(folder, projectName, features, cpp):
 
-    if cpp:
-        filename = Path(folder) / (projectName + '.cpp')
-    else:
-        filename = Path(folder) / (projectName + '.c')
-
-    file = open(filename, 'w')
-
-    main = ('#include <stdio.h>\n'
-            '#include "pico/stdlib.h"\n'
-            )
-    file.write(main)
+    template = jinja_env.get_template('main.txt')
+    mapping = dict(includes="", defines="", initialisers="")
 
     if (features):
-
         # Add any includes
+        includes = []
         for feat in features:
             if (feat in features_list):
-                o = '#include "' +  features_list[feat][H_FILE] + '"\n'
-                file.write(o)
+                includes.append(f'#include "{features_list[feat][H_FILE]}"')
```

Comment:
Just a personal viewpoint, but if this was me I'd make `includes` just be a list of filenames (which e.g. gives more flexibility for additional processing later), and then in the template do e.g.
```
{% for include in includes %}
include "{{include}}"
{% endfor %}
```
(I'm not familiar with Jinja, so I may not have got the syntax right.)

Reply:
Makes sense.